### PR TITLE
[ACS-777] Email notification for digital workspace

### PR DESCRIPTION
**Pull request by dhrn**
REST APIs that send email to use an optional client name. The client name is used to lookup the custom email template and its related assets based on the registered configuration. 

https://alfresco.atlassian.net/browse/ACS-777

depends https://github.com/Alfresco/rest-api-explorer/pull/108

### CODE REVIEW

File: .gitignore
Comment (alandavis) [inline .gitignore:40]: Normally the alf_data directory is not seen. Is this really needed for everyone? Looks like it may be because of local config.

**Review by alandavis — COMMENTED** [review]
I have had a very fast scan of these changes. However given the amount of new code, I think you really need to go through the PR in detail with someone in your team explaining what it is doing. There is one thing as a reviewer I would want to be sure about.

There should be checks that the new request parameters are not simply used 'as is' in the email. To do so would open us up to security issues such as returning sensitive information on the server or hosting an attack.

Note: As much as don't like the style that does not put a space after key words in Java such as "if(..." "for(...", I can see that that was the style in the files already. You have made the correct choice in not reformatting, as part of this PR. If we were to reformat (please don't) it would be done in a separate PR.

File: repository/src/main/java/org/alfresco/service/cmr/invitation/Invitation.java
Comment (jamalkm) [inline repository/src/main/java/org/alfresco/service/cmr/invitation/Invitation.java:105]: Code standard: It looks like you are using tabs rather than spaces

File: repository/src/main/java/org/alfresco/service/cmr/invitation/InvitationService.java
Comment (jamalkm) [inline repository/src/main/java/org/alfresco/service/cmr/invitation/InvitationService.java:152]: Same as my first comment; There is no need for `workspacePath` parameter.

File: repository/src/main/resources/alfresco/application-context-highlevel.xml
Comment (jamalkm) [inline repository/src/main/resources/alfresco/application-context-highlevel.xml:47]: What is the reason of moving this context file here?

File: repository/src/main/resources/alfresco/client/config/repo-clients-apps.properties
Comment (jamalkm) [inline repository/src/main/resources/alfresco/client/config/repo-clients-apps.properties:24]: You don't `sharedLinkBaseUrl` property for  the site invite.

**Review by jamalkm — CHANGES_REQUESTED** [review]

File: repository/src/main/resources/alfresco/application-context-highlevel.xml
Comment (dhrn) [reply repository/src/main/resources/alfresco/application-context-highlevel.xml:47]: This fixes the minimal context run test. this bean is having a couple of dependencies that are not required for the minimal run. so I moved the inviting service here.

**Review by eromano — APPROVED** [review]

**Review by jamalkm — COMMENTED** [review]
The changes look good to me. However, before I approve, I was wondering if you have tested an upgrade scenario? We have to make sure whether a patch is required or not.

File: remote-api/src/main/java/org/alfresco/rest/api/impl/SiteMembershipRequestsImpl.java
Comment (jamalkm) [inline remote-api/src/main/java/org/alfresco/rest/api/impl/SiteMembershipRequestsImpl.java:341]: extra semicolon `;;`

**Comment by dhrn** [comment]
I have tested with the upgrade from 6.2.2.1 with the help of an external volume attached to the docker-compose file. I didn't face any issue while going with the latest dev version

**Review by jamalkm — APPROVED** [review]

**Pull request merged**
